File: 0328-odd-even-linked-list/0328-odd-even-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode oddEvenList(ListNode head) {

        if(head==null || head.next==null)
        {
            return head;
        }

        ListNode slow=head.next;;
        ListNode fast=head;
        ListNode slowHead=head.next;

       

        while(slow!=null && slow.next!=null)
        {
           fast.next=fast.next.next;
           fast=fast.next;

           slow.next=slow.next.next;
           slow=slow.next;
        }

       fast.next=slowHead;


       return head;
        
    }
}

File: 0328-odd-even-linked-list/0328-odd-even-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def oddEvenList(self, head: Optional[ListNode]) -> Optional[ListNode]:

        if head==None or head.next==None:
            return head
        



        slow=head.next
        slowhead=head.next
        fast=head

        while fast.next!=None and slow.next!=None:
            fast.next=fast.next.next
            fast=fast.next

            slow.next=slow.next.next
            slow=slow.next
        fast.next=slowhead

        return head
        